text/00110-compatibility-strategy.md

```diff
@@ -0,0 +1,364 @@
+
+<br>
+
+# Drawbacks
+
+- Dramatically increases the duration of our integration tests suite.
+- Requires manually editing the framework compatibility test file.
+
+    > This isn't necessarily a drawback, as it provides a clear paper trail of all our breakages.
+
+# Rationale and Alternatives
+
+I'm not seeing any substantially different design approaches. Not matter what, to implement these validations, we have to implement the `run_integration_tests_perm` function, which is the bulk of the work. Any other approach would probably result in an incomplete solution. 
+
+This design provides a robust mechanism that should serve us well for the forseeable future. It covers all our compatibility promises without the need to add specific (functionality) tests.
+
+# Adoption Strategy
+
+Adoption of this strategy should be rather seamless for developers. Like we already mentioned, the changes a developer would need to incorporate, will be guided by failing tests. Hopefully, the process will not require any memorization (or even documentation). 
+
+Having said that, this RFC can be a good reference documentation that provides rational for every step in the process.
+
+# Unresolved questions
+
+There are some scenarios which are still a bit vague with regards to what should be our expected outcome.
+
+## Accidental (In)Compatibility
+
+Imagine a developer pushes some code that changes the `manifest.json`, without bumping the protocol version. Also imagine that this change is compatible with older CLI versions (maybe just an added property), meaning that all integration tests pass. Actually, you don't have to imagine, here is a real life example: [custom docker files support](https://github.com/aws/aws-cdk/pull/5652). Note that this feature only requires a new CLI to support a **new** feature, while all previous functionality remains intact.
+
+With this proposed mechanism, its very likely this code will be released without performing a version bump.
+
+In contrast, imagine the reverse, what if we push code that contains a version bump, making the framework incompatible, while in reality, existing functionality is not broken?
+
+This begs the question, when do we actually want a version bump? On any change to the `manifest.json`? Perhaps only on changes that break the CLI? Maybe we should require CLI upgrades only when detecting usages of specific features?
+
+The reason I call this *accidental*, is because this mechanism tries to shift most of the compatibility concerns away from humans, and into the hands of the machine. While this is generally a good approach, it does mean that we might be relaxing our brains a bit too much, causing some affects that possible wouldn't have happened in a more manual state of mind.
+
+## Avoiding Regressions
```

Comment:
Why is this out of scope? Feels like it should be included

Reply:
Because regressions aren't necessarily related to CLI vs Framework compatibility. However, it does use the same mechanisms...so it might be worth adding actually. I'll add it.